lib/Web/Main.hs

```diff
@@ -4,18 +4,75 @@
 
 module Web.Main
   ( runServer
+  , RunServerException (..)
   ) where
 
 import Backend.Commands (runCommand)
 import Config
+import Control.Exception (Exception, throw)
+import Data.Bifunctor (Bifunctor(first))
+import Data.Either (partitionEithers)
 import Data.Proxy
 import Network.Wai.Handler.Warp (run)
+import Options.Applicative
 import Servant.Server (serve)
+import System.Environment (lookupEnv)
+import Text.Interpolation.Nyan
+import Text.Read (readEither)
 import Web.API (API)
 import Web.Server
 
+data ServerOptions = ServerOptions { serverPort :: Maybe Int }
+
+data RunServerException
+  = RunServerIncorrectPort Int
+  | RunServerNoPortSpecified
+  | RunServerEnvVarParseFail String
+  deriving stock (Eq)
+
+instance Show RunServerException where
+  show (RunServerIncorrectPort port) = "Port number should be greater then 0. Your port number is: " ++ show port
+  show RunServerNoPortSpecified = "No port where not specified. Can be set by envirment varieble \"COFFER_SERVER_PORT\" or by \"--port=$port_num\" cmd argument."
+  show (RunServerEnvVarParseFail msg) = "Can't parse port specified by env var \"COFFER_SERVER_PORT\": " ++ msg
```

Comment:
One error message says "environment variable" and the other says "env var". Let's be consistent and use "environment variable" in both.

```suggestion
  show (RunServerEnvVarParseFail msg) = "Can't parse port specified by the environment variable \"COFFER_SERVER_PORT\": " ++ msg
```

Reply:
done
But left 'env var' in test msg-s (hope it's acceptable, that are for our eyes :> )